spark/v3.5/build.gradle

```diff
@@ -59,6 +59,7 @@ project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}") {
     implementation project(':iceberg-parquet')
     implementation project(':iceberg-arrow')
     implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}:${libs.versions.scala.collection.compat.get()}")
+    implementation("org.apache.datasketches:datasketches-java:${libs.versions.datasketches.get()}")
```

Comment:
Does Spark by any chance ship this? Do we have to worry about conflicts?

Reply:
Thanks for catching this. Looks like sql/catalyst uses the same [library](https://github.com/apache/spark/blob/master/pom.xml#L216).
Should we shade this here and pin the version?

Reply:
I have shaded the library

Reply:
Makes sense.